src/main/java/io/github/cottonmc/resources/ItemResourceType.java

```diff
@@ -0,0 +1,98 @@
+package io.github.cottonmc.resources;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import io.github.cottonmc.cotton.Cotton;
+import io.github.cottonmc.cotton.registry.CommonItems;
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class ItemResourceType implements ResourceType {
+    protected final String name;
+    protected final Set<String> affixes = new HashSet<>();
+
+    public ItemResourceType(String name) {
+        this.name = name;
+    }
+
+    public ItemResourceType(String name, String... affixes) {
+        this.name = name;
+        withAffixes(affixes);
+    }
+
+    @Override
+    public String getDomain() {
+        return name;
+    }
+
+    @Override
+    public boolean governs(String itemName) {
+        if (itemName.equals(name) && affixes.contains("")) return true;
```

Comment:
Is this a placeholder? because I am pretty sure that this will miss all the block items. And all affix items?

Reply:
Nevermind, affixes are apparently handled directly in ResourceType. But what about the block items?

Reply:
blocks are also items

Reply:
ah, I see. You also check for blocks in governs. I think this explains things.

Reply:
ItemResourceType is for groups of resources that don't have any associated blocks. For coal coke, moving forward, we want to add a generalized type, but that can be another, less-controversial PR.